test/hotspot/jtreg/serviceability/attach/LongArgTest.java

```diff
@@ -0,0 +1,161 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Tests that long arguments of attach operation are not truncated
+ * @bug 8334168
+ * @library /test/lib
+ * @modules jdk.attach/sun.tools.attach
+ * @run main LongArgTest
+ */
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.io.IOException;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.tools.attach.HotSpotVirtualMachine;
+
+import jdk.test.lib.apps.LingeredApp;
+
+public class LongArgTest {
+
+    public static void main(String[] args) throws Exception {
+        LingeredApp app = null;
+        try {
+            app = LingeredApp.startApp();
+
+            test(app).run();
+
+            test(app)
+                .withLongValue()
+                .run();
+
+            test(app)
+                .withSuperLongValue()
+                .run();
+
+        } finally {
+            LingeredApp.stopApp(app);
+        }
+
+    }
+
+    // For simplicity, the test uses internal HotSpotVirtualMachine,
+    // sets/gets "HeapDumpPath" flag value (string flag, not validated by JVM).
+    private static class Test {
+        private LingeredApp app;
+        private String flagName = "HeapDumpPath";
+        private String flagValue = generateValue(5);
+
+        Test(LingeredApp app) {
+            this.app = app;
+        }
+
+        // Value length exceeds 1K.
+        Test withLongValue() {
+            flagValue = generateValue(1024 + 1);
+            return this;
+        }
+
+        // Value length exceeds 3K (1K * 3 args).
+        Test withSuperLongValue() {
+            flagValue = generateValue(3 * 1024 + 1);
+            return this;
+        }
+
+        void run() throws Exception {
+            System.out.println("======== Start ========");
+
+            HotSpotVirtualMachine vm = (HotSpotVirtualMachine)VirtualMachine.attach(String.valueOf(app.getPid()));
+
+            if (setFlag(vm)) {
+                String actualValue = getFlag(vm);
+
+                if (!flagValue.equals(actualValue)) {
+                    String msg = "Actual values is different: "
+                                + (actualValue == null
+                                    ? "null"
+                                    : ("size is " + actualValue.length() + " instead of " + flagValue.length()));
```

Comment:
Q: What if value is different but size is the same?
I understand the probability is very low but this is still confusing.

Reply:
I updated the code to check if the value was truncated